### PR TITLE
Python. Fix ConnectionResetError in py/selenium/webdriver

### PR DESCRIPTION
**Pull request by rth**
Closes https://github.com/SeleniumHQ/selenium/issues/6438

This implements the solution proposed in https://github.com/SeleniumHQ/selenium/issues/6438#issuecomment-425709445 by @cgoldberg

- [X] By placing an `X` in the preceding checkbox, I verify that I have signed the [Contributor License Agreement](https://github.com/SeleniumHQ/selenium/blob/master/CONTRIBUTING.md#step-6-sign-the-cla)

---
This change is [Reviewable](https://reviewable.io/reviews/seleniumhq/selenium/6480)

### CODE REVIEW

File: py/selenium/webdriver/common/utils.py [hunk 1]
```diff
@@ -102,10 +102,16 @@ def is_connectable(port, host="localhost"):
      - port - The port to connect.
     """
     socket_ = None
+
+    if sys.version_info > (3, 4):
+        expected_exceptions = (socket.error, ConnectionResetError)
+    else:
+        expected_exceptions = (socket.error,)
+
```
Comment (lmtierney) [inline py/selenium/webdriver/common/utils.py]: You can throw this at the top with:

https://github.com/SeleniumHQ/selenium/blob/0d40dcd64295047d8669e688927a414d399e712e/py/selenium/webdriver/common/utils.py#L24-L28

Comment (rth) [reply py/selenium/webdriver/common/utils.py]: Thanks for the review! Sure -- done.

**Comment by cgoldberg** [comment]
👍

**Comment by PetterS** [comment]
Any change this PR can be merged? I would love to be able to use a version of Selenium more recent than 2017.

**Comment by mauritsvanrees** [comment]
In the Plone project, we were often getting ConnectionResetErrors in one or more of our 130 robot test cases on Python 3. Since about a week we are using a [patched version of the above fix](https://github.com/plone/plone.app.robotframework/pull/108/files) in our tests on [Jenkins](https://jenkins.plone.org) and it seems to hold up fine, both on Python 2 (no real changes) and Python 3.

**Comment by PetterS** [comment]
Does anyone have an idea when a version of Selenium with this fix will be released?

**Comment by AutomatedTester** [comment]
You can use it now in the 4.0 alpha release

**Comment by PetterS** [comment]
OK thanks, I think I'll wait for a stable release on pypi though.

**Pull request merged**